### PR TITLE
feat: Display images in the app

### LINKED ISSUE #762: Showing image in the response to queries.
**Actual Behaviour**

We are not showing images even when there is image URL present in the response from the server.

**Expected Behaviour**

Just like IOS and Web Clients we should show the image in the answer if image URL is present in the response.

**Screenshots of the issue**

screen shot 2017-06-22 at 3 18 57 pm

**Would you like to work on the issue?**

Yes, already working on this.

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #762 

Changes: Display image queries in the app and on clicking the image display chrome tab to show enlarged images
 
screenshot_20180627-140353 screenshot_20180627-140229

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt [hunk 1]
```diff
@@ -7,6 +7,7 @@ import org.fossasia.susi.ai.helper.Constant
 import org.fossasia.susi.ai.rest.responses.susi.Datum
 import org.fossasia.susi.ai.rest.responses.susi.SusiResponse
 import timber.log.Timber
+import java.util.regex.Pattern
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt]: Unused

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java [hunk 2]
```diff
@@ -220,6 +222,9 @@ public void onDetachedFromRecyclerView(RecyclerView recyclerView) {
             case VIDEOPLAY:
                 view = inflater.inflate(R.layout.youtube_video, viewGroup, false);
                 return new YoutubeVideoViewHolder(view, clickListener);
+            case IMAGE:
+                view = inflater.inflate(R.layout.image_holder,viewGroup,false);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Space after comma

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        ButterKnife.bind(this, itemView);
+    }
+
+    public void setView(ChatMessage model) {
+        this.model = model;
+
+        if (model != null) {
+            imageURL = model.getContent();
+            try {
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Extra line

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                Picasso.with(itemView.getContext()).load(imageURL).
+                        placeholder(R.drawable.ic_susi)
+                        .into(imageView);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Follow style guideline

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        } else if (!model.isPositiveRated() && model.isNegativeRated()) {
+            rateSusiSkill(Constant.POSITIVE, model.getSkillLocation(), itemView.getContext());
+            setRating(false, false);
+            thumbsDown.setImageResource(R.drawable.thumbs_down_outline);
+        }
+    }
+
+    // a function to rate the susi skill
+    private void rateSusiSkill(final String polarity, String locationUrl, final Context context) {
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Extra line

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        final Map<String, String> susiLocation = ParseSusiResponseHelper.Companion.getSkillLocation(locationUrl);
+
+        if (susiLocation.size() == 0) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Use isEmpty

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            return;
+        }
+
+        Call<SkillRatingResponse> call = new ClientBuilder().getSusiApi().rateSkill(susiLocation.get("model"),
+                susiLocation.get("group"), susiLocation.get("language"), susiLocation.get("skill"), polarity);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Style guideline

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+                            break;
+                        default:
+                    }
+                    Toast.makeText(context, context.getString(R.string.error_rating), Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<SkillRatingResponse> call, Throwable t) {
+                t.printStackTrace();
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Timber

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+                            setRating(false, false);
+                            break;
+                        default:
+                    }
+                    Toast.makeText(context, context.getString(R.string.error_rating), Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<SkillRatingResponse> call, Throwable t) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Parameter name

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+
+    // function to set the rating in the database
+    private void setRating(boolean what, boolean which) {
+        Realm realm = Realm.getDefaultInstance();
+        realm.beginTransaction();
+        if (which) {
+            model.setPositiveRated(what);
+        } else {
+            model.setNegativeRated(what);
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Which? What?

**Comment by batbrain7** [comment]
@iamareebjamal changes done

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java [hunk 11]
```diff
@@ -233,10 +238,13 @@ public int getItemViewType(int position) {
         if (item.getId() == -404) return DOTS;
         else if (item.getId() == -405) return NULL_HOLDER;
         else if (item.isDate()) return DATE_VIEW;
+        else if ((item.getContent().endsWith(".jpg") || (item.getContent().endsWith(".png"))))
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: This isn't lisp. Why are there so many parentheses where it just could be

```Java
item.getContent().endsWith(".jpg") || item.getContent().endsWith(".png")
```
```

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+    public ImageView imageView;
+    @BindView(R.id.timestamp)
+    public TextView timeStamp;
+    @BindView(R.id.thumbs_up)
+    public ImageView thumbsUp;
+    @Nullable
+    @BindView(R.id.thumbs_down)
+    public ImageView thumbsDown;
+    private ChatMessage model;
+    String imageURL;
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/ImageViewHolder.java]: Codacy has said something

**Pull request merged**